pkg/operator/mon/health.go

```diff
@@ -143,9 +143,43 @@ func (c *Cluster) checkHealth() error {
 		return nil
 	}
 
+	// check if there are more than two mons running on the same node, failover one mon in that case
+	ret, err := c.checkMonsUsed()
+	if ret || err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (c *Cluster) checkMonsUsed() (bool, error) {
+	nodesUsed := map[string]struct{}{}
+	for name, node := range c.mapping.Node {
+		// TODO Check if node is still valid else continue with rebalance
+
+		// when the node is already in the list we have more than one mon on that node
+		if _, ok := nodesUsed[node.Name]; ok {
+			// get list of available nodes for mons
+			availableNodes, _, err := c.getAvailableMonNodes()
```

Comment:
seems like this rebalance logic would also be used if the other check above decides a node is not valid, but it could be refactored later